src/editor/dag.rs

```diff
@@ -177,6 +177,23 @@ impl std::fmt::Display for Insertable {
     }
 }
 
+/// A datastructure that stores the current tree before performing some editing.
```

Comment:
This doc comment doesn't quite describe what `Snapshot` does - a `Snapshot` is the entire representation of a single edit of the tree.  Something like
```rust
/// A representation of a single edit, along with the cursor locations around it
```
would be better, I think.  This is pretty nitpicky though...

Reply:
No problem!